bin/dune_init.ml

```diff
@@ -519,7 +519,19 @@ module Component = struct
             [ File.make_text (Path.parent_exn opam_file) (Path.basename opam_file) "" ]
           | Esy -> [ File.make_text dir "package.json" "" ]
         in
-        { dir; files = dune_project_file dir opts :: package_files }
+        let default_files =
+          let exists =
+            Path.exists (Path.relative dir ".ocamlformat")
+            && Path.exists (Path.relative dir ".gitignore")
+          in
+          if exists
+          then []
+          else
+            [ File.make_text dir ".ocamlformat" ""
```

Comment:
It would be useful to set the version, if ocamlformat is installed at this time.

```
version = <output of ocamlformat --version>
```

Reply:
This could be annoying if you create a project, then a local switch and that switch then uses a different ocamlformat.

Reply:
Yeah, we are aiming for the basics to get started here, it's fine to not lock things doing at this point. Later if the user needs a specific version they can always set it. We are not making an irreversible decision here.

Reply:
The version mismatch is explicitly annoying but without specifying the version, the code might be formatted differently on the different switches.

We recommend always specifying a version in `.ocamlformat` files.

Reply:
If we specify a version in the `.ocamlformat` file, then we upgrade to a newer version of ocamlformat, it will refuse to run because of the mismatched version. Then the options are to either change the version number in the file and reformat the code, or to downgrade ocamlformat and keep the formatting unchanged.

If we don't specify a version and we upgrade to a newer version, it will run and reformat the code, in which case we will notice and either accept the reformat and check it in, or downgrade ocamlformat and keep the formatting unchanged.

I don't see any significant difference between the options.

Reply:
In the second scenario, it might be hard to not accept the new formatting as new code have been written. You'd have to remember the previous version of ocamlformat you used and install that.
The change in formatting might also be unnoticed and pollute the history and that might be frequent when switching back and forth between two switches or working on other projects that use different versions.
Formatting check in CI needs to know the version.
As this can affect the entire source tree, the ocamlformat team thinks that the explicit version check is better.

Reply:
Then I suppose that adding the version is the more logical choice if it's recommended by the `ocamlformat` team. I'll try to push a commit as soon as possible to add this feature.

Thanks you all for your time.